### PR TITLE
Fix dropdown blinking ux i& add trim option

### PR DESCRIPTION
**Pull request by yidea**
Changes: 
- Solve remote fetch dropdown blinking ux issue
- Add support for trim options 
- Update unit tests and demo page

@ryan-roemer @alexlande

### CODE REVIEW

File: dist/typeahead.js [hunk 1]
```diff
@@ -173,8 +175,11 @@
                 }
             },
             _getRegex: function(patterns) {
-                var patternsEscaped = [];
+                var self = this, patternsEscaped = [];
```
Comment (ryan-roemer) [inline dist/typeahead.js:178]: GLOBAL: Newline + proper indentation for each variable in single `var` declaration.

File: dist/typeahead.js [hunk 2]
```diff
@@ -606,7 +611,9 @@
                 if (suggestions.length < this.limit && this.transport) {
                     cacheHit = this.transport.get(query, processRemoteData);
                 }
-                !cacheHit && cb && cb(suggestions);
+                if (cacheHit === undefined) {
```
Comment (ryan-roemer) [inline dist/typeahead.js]: Switch to:

``` js
if (typeof cacheHit === "undefined" && cb) {
    cb(suggestions);
}
```

File: dist/typeahead.js [hunk 3]
```diff
@@ -1003,7 +1015,7 @@
...
+            }).on("focused", this._openDropdown).on("blured", this._closeDropdown).on("blured", this._setInputValueToQuery).on("enterKeyed tabKeyed", this._handleSelection).on("queryChanged", this._clearHint).on("queryChanged", this._getSuggestions).on("queryChangedEmpty", this._clearSuggestions).on("whitespaceChanged", this._updateHint).on("queryChanged whitespaceChanged", this._openDropdown).on("queryChanged whitespaceChanged", this._setLanguageDirection).on("escKeyed", this._closeDropdown).on("escKeyed", this._setInputValueToQuery).on("tabKeyed upKeyed downKeyed", this._managePreventDefault).on("upKeyed downKeyed", this._moveDropdownCursor).on("upKeyed downKeyed", this._openDropdown).on("tabKeyed leftKeyed rightKeyed", this._autocomplete);
```
Comment (ryan-roemer) [inline dist/typeahead.js:1018]: Can you break the chains across newlines? Hard to read (I know this is what it started as).

**Comment by ryan-roemer** [comment]
A couple of comments, but generally LGTM. Assigning to @alexlande to test out and complete review. /cc @yidea

File: dist/typeahead.js (hunk 3 above)
Comment (yidea) [inline dist/typeahead.js:1018]: @ryan-roemer, this dist/typeahead.js is auto-generated by grunt-contrib-uglify when running grunt build, and based on the option beatify: true. I was playing with the beatify options as http://lisperator.net/uglifyjs/codegen listed, but seems there's no direct control on the chaining part, do you have any good solution on this?

Comment (yidea) [inline dist/typeahead.js:1018]: output sourcemap in grunt-contrib-uglify task as an option?

Comment (ryan-roemer) [inline dist/typeahead.js:1018]: If it's auto-generated, then just leave it as is!

**Comment by yidea** [comment]
Fix for code review: 
- add grunt-contrib-uglify source map support (grunt build to generate), update "grunt-contrib-uglify" to 0.2.7
- fix code review issue (global var formating, typeof undefined)

@ryan-roemer @alexlande

File: dist/typeahead.js (hunk 3 above)
Comment (ryan-roemer) [inline dist/typeahead.js:1018]: Why are we doing sourcemaps?

Comment (yidea) [inline dist/typeahead.js:1018]: thought you want the combined typeahead.js more readable, so added sourcemaps, a bad idea?

**Comment by alexlande** [comment]
LGTM @yidea (pending confirmation from @ryan-roemer that the source map should be there).

**Pull request merged**